src/main/java/com/atlassian/db/replica/internal/ReplicaStatement.java

```diff
@@ -331,6 +334,7 @@ public int executeUpdate(String sql, String[] columnNames) throws SQLException {
     @Override
     public boolean execute(String sql, int autoGeneratedKeys) throws SQLException {
         checkClosed();
+        //all execute(***) cases might be executed by readSQL, but we treat them as Write, this might be optimization for future
```

Comment:
Please create an issue. Our application would probably have no gain (we use mostly prepared statements and use `#execute` for locks only), but it can be a big change for other apps.

Reply:
https://github.com/atlassian-labs/db-replica/issues/53